Lib/csv.py

```diff
@@ -364,28 +364,35 @@ def _guess_delimiter(self, data, delimiters):
         try and evaluate the smallest portion of the data possible, evaluating
         additional chunks as necessary.
         """
+        from collections import Counter, defaultdict
 
         data = list(filter(None, data.split('\n')))
 
-        ascii = [chr(c) for c in range(127)] # 7-bit ASCII
+        ascii = {chr(c) for c in range(127)} # 7-bit ASCII
```

Comment:
This is constant data, it makes sense to compute it once at module-level, e.g. `_ASCII_CHARS = frozenset(map(chr, range(127))) # 7-bit ASCII`.
```suggestion
        ascii = frozenset(map(chr, range(127))) # 7-bit ASCII
```